tools/reflection_generator/java_class.py

```diff
@@ -322,6 +329,7 @@ def ExtractMethods(self, java_content):
           method_name, method_return, method_params,
           method_annotation, method_doc)
       self._methods.append(method)
+      self._need_default_constructor = False
```

Comment:
A class doesn't need default constructor just because it has a static method? We should check the annotation noInstance instead.